### DISCUSSION TITLE
Missing Search results for specific languages

### DISCUSSION THREAD

**Discussion by VictorRomanov1986** [microsoft-search/pnp-modern-search#3763] [Q&A]
**Version used**
Ex: 4.11.1

**Describe the bug**
I have a page, which display all news with filters. One of the filters is by Location
![image](https://github.com/microsoft-search/pnp-modern-search/assets/42931275/418d86a7-bfc2-4156-99a9-185dfb969100)
(this is a managed metadata property mapped with RefinableString06)
On the screen shot above there is a Taiwan option with 5 results.
But when I apply this filter, it starts to show only 2 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/42931275/08da9db4-aee1-46a2-a786-02d2c380f9c2)
But at the same time in the Site Pages library there are 5 pages, promoted as News.
Fact - if I remove the text in one of the missing pages - it starts to appear in Search results
Can you please suggest what could be the problem?

**Comment by kasperbolarsen** [comment]
Hi, a few options you can check:
Please check if the Results web part has "remove duplicates " enabled,  it should be off.

Another option is to execute a query like :RefinableString06:"Taiwan: in either the classic search or SP Editor/SharePoint Search Query tool 

The totals returned from the API are not really trustworthy

**Comment by VictorRomanov1986** [comment]
> Hi, a few options you can check: Please check if the Results web part has "remove duplicates " enabled, it should be off.
> 
> Another option is to execute a query like :RefinableString06:"Taiwan: in either the classic search or SP Editor/SharePoint Search Query tool
> 
> The totals returned from the API are not really trustworthy

Thanks for the advise
"Remove duplicates" is off
I tried to do the query using Query Builder, and it is returning all posts
![image](https://github.com/microsoft-search/pnp-modern-search/assets/42931275/287a9fb6-e6ff-49f1-8b64-d5a0cd6784a1)

**Comment by kasperbolarsen** [comment]
Ok, so there must be something wrong with your Query. 
Can you share the query from the network tab in Dev Tools, search for "postquery" in the output.

**Comment by VictorRomanov1986** [comment]
{
  "__metadata": {
    "type": "Microsoft.Office.Server.Search.REST.SearchRequest"
  },
  "ClientType": "PnPModernSearch",
  "Properties": {
    "results": [
      {
        "Name": "EnableDynamicGroups",
        "Value": {
          "BoolVal": true,
          "QueryPropertyValueTypeIndex": 3
        }
      },
      {
        "Name": "EnableMultiGeoSearch",
        "Value": {
          "BoolVal": true,
          "QueryPropertyValueTypeIndex": 3
        }
      },
      {
        "Name": "ClientFunction",
        "Value": {
          "StrVal": "PnPSearchWebPart",
          "QueryPropertyValueTypeIndex": 1
        }
      },
      {
        "Name": "ListId",
        "Value": {
          "StrVal": "5b23d736-67f9-4c17-a70e-53830d796941",
          "QueryPropertyValueTypeIndex": 1
        }
      },
      {
        "Name": "ListItemId",
        "Value": {
          "StrVal": "8",
          "QueryPropertyValueTypeIndex": 1
        }
      }
    ]
  },
  "Querytext": "",
  "EnableQueryRules": false,
  "QueryTemplate": " (ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C411800BF4B5111716AE34DADCB16AFFDDF92CB* OR ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C4118002A50BFCFB7614729B56886FADA02339B00DEF783C5CA98481EAFCB246E2A3E7163*) AND PromotedState=2 AND (SPTranslationLanguage:{CurrentDisplayLanguage} OR (NOT SPTranslatedLanguages:{CurrentDisplayLanguage} AND NOT SPIsTranslation:true))",
  "SourceId": "8413cd39-2156-4e00-b54d-11efd9abdb89",
  "EnableNicknames": false,
  "EnablePhonetic": false,
  "Culture": 1033,
  "TimeZoneId": 75,
  "Refiners": "FirstPublishedDate(discretize=manual/2023-05-14T08:29:45.799Z/2024-02-14T09:29:45.800Z/2024-04-14T08:29:45.800Z/2024-05-07T08:29:45.801Z/2024-05-13T08:29:45.801Z/2024-05-14T08:30:45.801Z),RefinableString00(deephits=1000000),RefinableString05(deephits=1000000),RefinableString01(deephits=1000000),RefinableString06(deephits=1000000),RefinableString02(deephits=1000000),RefinableString04(deephits=1000000)",
  "RefinementFilters": {
    "results": [
      "and(RefinableString00:or(GP0|#e5877cc8-7500-468c-a28a-c93158b5d2ee,L0|#0e5877cc8-7500-468c-a28a-c93158b5d2ee),RefinableString01:or(GP0|#ae65aa1a-6649-4d76-adb0-8a4cc2df29a9,L0|#0ae65aa1a-6649-4d76-adb0-8a4cc2df29a9),RefinableString05:or(GP0|#87b99401-22c1-41a1-a5e8-9ba9b26d735a,L0|#087b99401-22c1-41a1-a5e8-9ba9b26d735a),RefinableString06:or(GP0|#ed69b44d-0b27-4967-a4b6-165d824d6105,L0|#0ed69b44d-0b27-4967-a4b6-165d824d6105))"
    ]
  },
  "RowLimit": 12,
  "StartRow": 0,
  "TrimDuplicates": false,
  "SortList": {
    "results": [
      {
        "Property": "FirstPublishedDate",
        "Direction": 1
      }
    ]
  },
  "SelectProperties": {
    "results": [
      "Author",
      "AuthorOWSUser",
      "DisplayUser",
      "Description",
      "FirstPublishedDate",
      "PictureThumbnailURL",
      "RefinableString00",
      "RefinableString01",
      "RefinableString02",
      "RefinableString03",
      "RefinableString04",
      "RefinableString05",
      "RefinableString06",
      "ViewCount"
    ]
  },
  "HitHighlightedProperties": {
    "results": []
  },
  "ReorderingRules": {
    "results": []
  }
}

**Comment by kasperbolarsen** [comment]
One thing that I noticed it that the query said PromotedState=2
Can you verify that all 5 pages actually are published?

**Comment by VictorRomanov1986** [comment]
> One thing that I noticed it that the query said PromotedState=2 Can you verify that all 5 pages actually are published?

Yes, All 5 pages are Published and Promoted as News

**Comment by kasperbolarsen** [comment]
OK, let's try next option,  try removing the SPTranslatedLangues and see what happens

**Comment by VictorRomanov1986** [comment]
Did not helped
![image](https://github.com/microsoft-search/pnp-modern-search/assets/42931275/1d124404-4377-4129-9d92-199bf2716796)
Morover, if I use not term id, but the label of the term, Search returning more results
![image](https://github.com/microsoft-search/pnp-modern-search/assets/42931275/3682ba81-4f58-4055-ba54-28cd042d74b0)
Maybe it is a SharePoint Classic Search issue?

**Comment by kasperbolarsen** [comment]
Any chance you can verify the term guid for the 5 items that shows up when you are searching for the term label "Taiwan". If they are not identical,  than it is a data quality issue. 

You don't have two "Taiwan" in the term store, right? I have seen that one at a previous occasion and it took forever to find the root cause.

**Comment by VictorRomanov1986** [comment]
> Any chance you can verify the term guid for the 5 items that shows up when you are searching for the term label "Taiwan". If they are not identical, than it is a data quality issue.
> 
> You don't have two "Taiwan" in the term store, right? I have seen that one at a previous occasion and it took forever to find the root cause.

There is only 1 term with Taiwan name. Moreover - this field is hidden for end-user and set with a default value

**Comment by kasperbolarsen** [comment]
And all 5 hits have same guid for that property? You should be able to see the value In the allmetadataowsdata ( can't remember the exact name)

**Comment by VictorRomanov1986** [comment]
> And all 5 hits have same guid for that property? You should be able to see the value In the allmetadataowsdata ( can't remember the exact name)

Yes, All pages have the same values

**Comment by kasperbolarsen** [comment]
I am running out of ideas. 
The best offer right now is to sign up for a half hour Teams meeting in the PnP Modern Search Office hours. 
The next free slot is 4 June, https://forms.office.com/pages/responsepage.aspx?id=xt4SFDF540m7kCquHMmhqrjqma_5-shDoiBP0hNegIRUMVhNN0ZDQ0FRWkVWT0lDUDBCOVdTR0xLMi4u
